From e7ebac95c85fa0f223541189babf42378d2b4096 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Thu, 20 Jan 2022 18:50:39 +0200
Subject: [PATCH] [SW-69581] common: remove ca53 cfg hdr from common code

Including device specific headers from common code is not allowed
anymore, and instead the device structures should be used.
Common code is refactored to use device structure instead of
cpu_ca53_cfg header.

Change-Id: I9f672fab379d5c1a0d8dc0696a7a76e2aa64bcdb
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/app/ppboot/ppboot.c b/app/ppboot/ppboot.c
index 7e8be10..48f1d1b 100644
--- a/app/ppboot/ppboot.c
+++ b/app/ppboot/ppboot.c
@@ -6,7 +6,6 @@
  */
 
 #include <string.h>
-#include <cpu_ca53_cfg_regs.h>
 #include <hl_boot_if.h>
 #include <bitops.h>
 #include <preboot.h>
@@ -21,6 +20,8 @@
 #include <boot_if.h>
 #include <pci.h>
 #include <hl.h>
+
+struct device_ca53_cfg ca53_cfg;
 
 __weak bool ppboot_is_valid_img_desc(struct boot_desc *desc)
 {
@@ -212,12 +213,12 @@
 	else
 		cpu_boot_dev_sts_clr(CPU_BOOT_DEV_STS_DEBUG_EN);
 
-	reg = regmap_readl(mmCPU_CA53_CFG_ARM_RST_CONTROL);
+	reg = regmap_readl(ca53_cfg.arm_rst_control_reg);
 	if (is_debug_en)
 		reg |= BIT(12);
 	else
 		reg &= ~BIT(12);
-	regmap_writel(reg, mmCPU_CA53_CFG_ARM_RST_CONTROL);
+	regmap_writel(reg, ca53_cfg.arm_rst_control_reg);
 }
 
 __weak void hl_init_pcie_phy_fw(void) {}
@@ -266,6 +267,7 @@
 	device_init_uart();
 	device_init_pci();
 	device_init_global_conf();
+	device_init_ca53_cfg();
 }
 
 __weak void ppboot_security_init(void) {}
diff --git a/arch/gaudi/arm/common.c b/arch/gaudi/arm/common.c
index bbb04a9..672fab0 100644
--- a/arch/gaudi/arm/common.c
+++ b/arch/gaudi/arm/common.c
@@ -23,6 +23,7 @@
 #include <pcie_dbi_regs.h>
 #include <pcie_dbi_masks.h>
 #include <cpu_timestamp_regs.h>
+#include <cpu_ca53_cfg_regs.h>
 #include <gic_regs.h>
 #include <nic0_phy_regs.h>
 
@@ -722,3 +723,8 @@
 	global_conf.preboot_ver_off_map = mmPREBOOT_VER_OFFSET;
 	global_conf.mmfuse_ver_off_map = mmFUSE_VER_OFFSET;
 }
+
+void device_init_ca53_cfg(void)
+{
+	ca53_cfg.arm_rst_control_reg = mmCPU_CA53_CFG_ARM_RST_CONTROL;
+}
diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index ad6612a..77bd7d5 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -26,6 +26,7 @@
 #include <pcie_aux_regs.h>
 #include <pcie_core_regs.h>
 #include <cpu_timestamp_regs.h>
+#include <cpu_ca53_cfg_regs.h>
 #include <cpu_if_regs.h>
 #include <cpu_if_masks.h>
 #include <preboot.h>
@@ -1291,3 +1292,8 @@
 	global_conf.preboot_ver_off_map = mmPREBOOT_VER_OFFSET;
 	global_conf.mmfuse_ver_off_map = mmFUSE_VER_OFFSET;
 }
+
+void device_init_ca53_cfg(void)
+{
+	ca53_cfg.arm_rst_control_reg = mmCPU_CA53_CFG_ARM_RST_CONTROL;
+}
diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 9fe5d2d..bd28d92 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -5,6 +5,7 @@
 
 #include <d0_psoc_global_conf_regs.h>
 #include <d0_cpu_timestamp_regs.h>
+#include <d0_cpu_ca53_cfg_regs.h>
 #include <d0_psoc_gpio0_regs.h>
 #include <d0_psoc_qspi_regs.h>
 #include <d0_pcie_aux_regs.h>
@@ -218,6 +219,11 @@
 	global_conf.mmfuse_ver_off_map = mmFUSE_VER_OFFSET;
 }
 
+void device_init_ca53_cfg(void)
+{
+	ca53_cfg.arm_rst_control_reg = mmD0_CPU_CA53_CFG_ARM_RST_CONTROL;
+}
+
 void lkd_dev_specific_setup(void)
 {
 	cpu_boot_dev_sts_set(CPU_BOOT_DEV_STS_FW_HARD_RST_EN);
diff --git a/arch/goya/arm/common.c b/arch/goya/arm/common.c
index e2ada22..12516b5 100644
--- a/arch/goya/arm/common.c
+++ b/arch/goya/arm/common.c
@@ -19,6 +19,7 @@
 #include <pcie_core_regs.h>
 #include <cpu_timestamp_regs.h>
 #include <pcie_db_rsv_regs.h>
+#include <cpu_ca53_cfg_regs.h>
 #include <common/pci_ids.h>
 
 #include <cpu_timestamp_regs.h>
@@ -381,3 +382,8 @@
 	global_conf.preboot_ver_off_map = mmPREBOOT_VER_OFFSET;
 	global_conf.mmfuse_ver_off_map = mmFUSE_VER_OFFSET;
 }
+
+void device_init_ca53_cfg(void)
+{
+	ca53_cfg.arm_rst_control_reg = mmCPU_CA53_CFG_ARM_RST_CONTROL;
+}
diff --git a/arch/greco/common/common.c b/arch/greco/common/common.c
index eb42f1b..c732c30 100644
--- a/arch/greco/common/common.c
+++ b/arch/greco/common/common.c
@@ -30,6 +30,7 @@
 #include <cpu_timestamp_regs.h>
 #include <cpu_if_regs.h>
 #include <cpu_if_masks.h>
+#include <cpu_ca53_cfg_regs.h>
 #include <preboot.h>
 #include <device.h>
 #include <uart.h>
@@ -660,3 +661,8 @@
 	global_conf.preboot_ver_off_map = mmPREBOOT_VER_OFFSET;
 	global_conf.mmfuse_ver_off_map = mmFUSE_VER_OFFSET;
 }
+
+void device_init_ca53_cfg(void)
+{
+	ca53_cfg.arm_rst_control_reg = mmCPU_CA53_CFG_ARM_RST_CONTROL;
+}
diff --git a/include/device.h b/include/device.h
index 0433699..23a73f4 100644
--- a/include/device.h
+++ b/include/device.h
@@ -78,14 +78,20 @@
 	u32 mmfuse_ver_off_map;
 };
 
+struct device_ca53_cfg {
+	u32 arm_rst_control_reg;
+};
+
 extern struct device_efuse efuse;
 extern struct device_uart uart;
 extern struct device_pci pci;
 extern struct device_i2c_s i2c_s;
 extern struct device_global_conf global_conf;
+extern struct device_ca53_cfg ca53_cfg;
 
 void device_init_efuse(void);
 void device_init_uart(void);
 void device_init_pci(void);
 void device_init_i2c_s(void);
 void device_init_global_conf(void);
+void device_init_ca53_cfg(void);
